src/v/utils/expiring_promise.h

```diff
@@ -115,7 +121,7 @@ class expiring_promise {
     }
 
     bool _available{false};
-    ss::shared_promise<T> _promise;
+    ss::promise<T> _promise;
```

Comment:
As I remember needed `.available()` method which is part of the promise. Later I figured that the method has a lag and returns false even when value was actually set so I added a boolean flag to explicitly track status. Now since the necessity in the method has gone it looks save to replace it back to promise.

Reply:
The shared promise can not be moved because of the flag, when moved available is reset and state is corrupted. I will split this into two commits

Reply:
@mmaslankaprv I'm not sure what you are referring to here. the question here for me is: is sharing a property of the expiring promise that needs to be preserved?

Reply:
oh ok now I'm looking at the commits and I see you split out the switch away from shared promise with a note in the commit message explaining that the sharing isn't used. thanks!